src/components/send/SendContainer.js

```diff
@@ -74,8 +75,10 @@ export function SendContainer({ match, location }) {
     }, [location.key])
 
     useEffect(() => {
-        dispatch(getBalance())
-    }, [])
+        if (id && actionStatus.GET_BALANCE?.success) {
```

Comment:
@Patrick1904 fix works alright, but not sure I understand the logic, may you explain?

Reply:
Yep there are two actions dispatched when the user lands fresh on the page: `GET_BALANCE` and `CHECK_ACCOUNT_AVAILABLE`. `GET_BALANCE` tends to finish **last**, which overrides the `localAlert` needed to show the validation from `CHECK_ACCOUNT_AVAILABLE`. So I'm just waiting for `GET_BALANCE.success` to be `true` before dispatching the account validation, that way it's the last action to finish.

Reply:
@Patrick1904 any reason not to await on `dispatch(getBalance())` then? Or maybe just not rely on `localAlert` for form validation and kick off these in parallel to avoid delay?

in any case I think we can merge, just let's make sure to have a tracking issue for cleaner fix

Reply:
@vgrichina yea so `GET_BALANCE` gets dispatched as part of account refresh when user lands on the page, that's why I removed it from the component because it was already being dispatched. I agree it's not a super clean solution. @marcinbodnar do you have any idea how to avoid this with localAlert?